src/lib/support/ReferenceCountedHandle.h

```diff
@@ -26,6 +26,10 @@ class ReferenceCountedHandle
 {
 public:
     explicit ReferenceCountedHandle(Target & target) : mTarget(target) { mTarget.Retain(); }
+    // TODO this is not ideal, while we prevent the Optional version of this from causing an issue we could be suppressing reall
+    // errors.
+
+    // NOLINTNEXTLINE(clang-analyzer-core.CallAndMessage): When ~Value() is called the mData.~T(); is not called
```

Comment:
TODO comment is a little off. That is because I am still hoping that we can figure something out in Optional.h to suppress this clang warning. Still seeing how we can do that